index.bs

```diff
@@ -2740,30 +4746,107 @@ partial interface MLGraphBuilder {
         - *input*: an {{MLOperand}}. The input 4-D tensor.
         - *options*: an optional {{MLResample2dOptions}}. The optional parameters of the operation.
-            - *mode*: an {{MLInterpolationMode}}. The interpolation algorithm used to fill the output tensor values.
-                If not set, it is assumed to be the *Nearest Neighbor* interpolation.
-            - *scales*: a sequence of {{float}} of length 2. Each value represents the scaling factor used to scale in each spatial dimensions of input, [scale_height, scale_width]. If not set, the values are assumed to be [1.0, 1.0].
-            - *sizes*: a sequence of {{unsigned long}} of length 2. The target sizes for each spatial dimensions of input, [size_height, size_width]. When the target sizes are specified, the *options.scales* argument is ignored as the scaling factor values are derived from the target sizes of each spatial dimension of input.
-            - *axes*: a sequence of {{unsigned long}} of length 2. The two consecutive dimensions of the input tensor to which the interpolation algorithm applies. The valid values in the sequence are [0, 1], [1, 2] or [2, 3]. When not specified, the sequence is assumed to be [2, 3].
 
     **Returns:** an {{MLOperand}}. The output 4-D tensor.
 </div>
 
-### The reshape() method ### {#api-mlgraphbuilder-reshape}
+{{MLResample2dOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLResample2dOptions>
+    : <dfn>mode</dfn>
+    ::
+        An {{MLInterpolationMode}} [=string=].
+        Specifies the interpolation algorithm used to fill the output tensor values.
+        The default value is `"nearest-neighbor"`, standing for *Nearest Neighbor* interpolation.
+
+    : <dfn>scales</dfn>
+    ::
+        A sequence of {{float}} of length 2.
+        Specifies the scaling factor in each spatial dimensions of the input: [scaleHeight, scaleWidth].
+        The default value is [1.0, 1.0].
+
+    : <dfn>sizes</dfn>
+    ::
+        A sequence of {{unsigned long}} of length 2.
+        Specifies the target sizes for each spatial dimensions of the input: [sizeHeight, sizeWidth]. When the target sizes are specified, the {{MLResample2dOptions/scales}} argument is ignored, since the scaling factor values are derived from the target sizes of each spatial dimension of the input.
+
+    : <dfn>axes</dfn>
+    ::
+        A sequence of {{unsigned long}} of length 2.
+        Specifies the two consecutive dimensions of the input tensor to which the interpolation algorithm applies. The valid values in the sequence are [0, 1], [1, 2] or [2, 3].
+        The default value is [2, 3].
+</dl>
+
+<details open algorithm>
+
+  <summary>
+    To <dfn for="MLGraphBuilder">check resample options</dfn> given |options|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. If |options|.{{MLResample2dOptions/mode}} [=map/exists=], and if its value is not one of `"nearest-neighbor"` or `"linear"`, return `false`.
```

Comment:
Unnecessary, taken care of by WebIDL (unless the intention is that there will be MLInterpolationMode values that aren't supported here)